### PR TITLE
Get timestamp from aws for non-json parsers

### PR DESCRIPTION
**Pull request by GAHila**

### CODE REVIEW

File: lib/fluent/plugin/in_cloudwatch_logs.rb [hunk 1]
```diff
@@ -134,13 +134,13 @@ def run
       end
     end
 
-    def emit(stream, event)
+    def emit(stream, event) 
+      time = (event.timestamp / 1000).floor
       if @parser
-        @parser.parse(event.message) {|time, record|
+        @parser.parse(event.message) {|parsed_time,record|
```
[the remark this answers is not in the document]
Comment (okkez) [reply lib/fluent/plugin/in_cloudwatch_logs.rb]: This change breaks compatibility. We can add a new option to keep compatibility or bump up major version (add release note for this incompatible change).

[the remark this answers is not in the document]
Comment (cosmo0920) [reply lib/fluent/plugin/in_cloudwatch_logs.rb]: > We can add a new option to keep compatibility or bump up major version (add release note for this incompatible change).

We should keep compatibility as mush as possible.
I think that adding new parameter is required.

[the remark this answers is not in the document]
Comment (GAHila) [reply lib/fluent/plugin/in_cloudwatch_logs.rb]: I assume we are talking about a config parameter, default preserve the current behavior and by setting it we can control how we get the timestamp, right? @cosmo0920 @okkez

[the remark this answers is not in the document]
Comment (okkez) [reply lib/fluent/plugin/in_cloudwatch_logs.rb]: Yes, we can choose to keep current behavior(default) or to get the timestamp from cloudwatch-logs event, if we add the new parameter.

[the remark this answers is not in the document]
Comment (GAHila) [reply lib/fluent/plugin/in_cloudwatch_logs.rb]: Just added the parameter and kept the current behavior by default @okkez @cosmo0920

File: lib/fluent/plugin/in_cloudwatch_logs.rb [hunk 2]
```diff
@@ -135,16 +135,19 @@ def run
     end
 
     def emit(stream, event)
-      if @parser
-        @parser.parse(event.message) {|time, record|
+        if @parser
+          @parser.parse(event.message) {|time,record|
+            if @use_aws_timestamp
```
Comment (gregoryguillou) [inline lib/fluent/plugin/in_cloudwatch_logs.rb]: If you intend to use use_aws_timestamp, you probably need to add it in the section with all the config_param by line 12 -27.  Your addition should be like:

```
 config_param :use_aws_timestamp, :bool, default: false
```

👍 for the change btw

Comment (GAHila) [reply lib/fluent/plugin/in_cloudwatch_logs.rb]: Good catch! Missed commit that I just added!

**Comment by GAHila** [comment]
@cosmo0920 @okkez @gregoryguillou, please take another look at latest changes done as suggested by the reviews. Appreciate it very much, thank you!!!  We would like to move along as this plugin is a tool we heavily use to get cloudwatch logs.

**Comment by gregoryguillou** [comment]
I'm no reviewer @GAHila. Just if that can help: I had rebase your change on master a few days back and I'm running the plugin with your patch for about 5 days now on our environment. That is super useful to rely on the variable, get the right timing and view the logs (with parser regexp) from our lambda in ElasticSearch.

File: test/plugin/test_in_cloudwatch_logs.rb [hunk 3]
```diff
@@ -235,6 +279,30 @@ def default_config
     EOC
   end
 
+  private
```
Comment (okkez) [inline test/plugin/test_in_cloudwatch_logs.rb]: Redundant `private`.

File: test/plugin/test_in_cloudwatch_logs.rb [hunk 4]
```diff
@@ -235,6 +279,30 @@ def default_config
...
+    #{aws_sec_key}
+    #{region}
+    #{endpoint}
+    format csv
+    keys time,message
+    time_key time
+  EOC
+  end
+
+  private
```
Comment (okkez) [inline test/plugin/test_in_cloudwatch_logs.rb]: Redundant `private`.

Comment (GAHila) [reply test/plugin/test_in_cloudwatch_logs.rb]: Corrected

File: test/plugin/test_in_cloudwatch_logs.rb (hunk 3 above)
Comment (GAHila) [reply test/plugin/test_in_cloudwatch_logs.rb]: Corrected

**Pull request merged**